Sources/StreamChat/Workers/ChannelListUpdater.swift

```diff
@@ -48,32 +59,33 @@ class ChannelListUpdater: Worker {
-                        }
-
-                        // We are going to clean those channels that are present in the both the local and remote query,
-                        // and that have not been synched nor watched. Those are outdated, can contain gaps.
-                        let cidsToClean = localInRemote.subtracting(updatedChannels)
-                        session.cleanChannels(cids: cidsToClean)
-
-                        // We are also going to keep track of the unwanted channels
-                        // Those are the ones that exist locally but we are not interested in anymore in this context.
-                        // In this case, it is going to query local ones not appearing in remote, subtracting the ones
-                        // that are already being watched.
-                        unwantedCids = localNotInRemote.subtracting(watchedAndSynchedChannelIds)
+                    initialActions: { _ in
+                        // Do we even need this any more? Since we not reset the query channels on first fetch.
+//                        guard let queryDTO = session.channelListQuery(filterHash: updatedQuery.filter.filterHash) else { return }
+//
+//                        let localQueryCIDs = Set(queryDTO.channels.compactMap { try? ChannelId(cid: $0.cid) })
+//                        let remoteQueryCIDs = Set(channelListPayload.channels.map(\.channel.cid))
+//
+//                        let updatedChannels = synchedChannelIds.union(watchedAndSynchedChannelIds)
+//                        let localNotInRemote = localQueryCIDs.subtracting(remoteQueryCIDs)
+//                        let localInRemote = localQueryCIDs.intersection(remoteQueryCIDs)
+//
+//                        // We unlink those local channels that are no longer in remote
+//                        for cid in localNotInRemote {
+//                            guard let channelDTO = session.channel(cid: cid) else { continue }
+//                            queryDTO.channels.remove(channelDTO)
+//                        }
+//
+//                        // We are going to clean those channels that are present in the both the local and remote query,
+//                        // and that have not been synched nor watched. Those are outdated, can contain gaps.
+//                        let cidsToClean = localInRemote.subtracting(updatedChannels)
+//                        session.cleanChannels(cids: cidsToClean)
+//
+//                        // We are also going to keep track of the unwanted channels
+//                        // Those are the ones that exist locally but we are not interested in anymore in this context.
+//                        // In this case, it is going to query local ones not appearing in remote, subtracting the ones
+//                        // that are already being watched.
+//                        unwantedCids = localNotInRemote.subtracting(watchedAndSynchedChannelIds)
```

Comment:
From my understanding, since we now remove channels from the query on first fetch, we probably don't need this logic? 🤔